common/omv_csi.c

```diff
@@ -459,222 +437,58 @@ int omv_csi_probe(omv_i2c_t *i2c) {
     for (size_t i=0; i<dev_count; i++) {
         omv_csi_t *csi = &csi_all[i];
+
         csi->power_pol = power_pol;
         csi->reset_pol = reset_pol;
         csi->chip_id =  dev_list[i].chip_id;
         csi->slv_addr = dev_list[i].slv_addr;
         csi->detected = true;
 
-        switch (dev_list[i].chip_id) {
-            #if (OMV_OV2640_ENABLE == 1)
-            case OV2640_ID:
-                if (omv_csi_set_clk_frequency(OMV_OV2640_CLK_FREQ) != 0) {
-                    return OMV_CSI_ERROR_TIM_INIT_FAILED;
-                }
-                init_ret = ov2640_init(csi);
-                break;
-            #endif // (OMV_OV2640_ENABLE == 1)
+        uint32_t clk_hz = 0;
+        sensor_init_t init_fun = NULL;
 
-            #if (OMV_OV5640_ENABLE == 1)
-            case OV5640_ID: {
-                int freq = OMV_OV5640_CLK_FREQ;
-                #if (OMV_OV5640_REV_Y_CHECK == 1)
-                if (HAL_GetREVID() < 0x2003) {
-                    // Is this REV Y?
-                    freq = OMV_OV5640_REV_Y_FREQ;
-                }
-                #endif
-                if (omv_csi_set_clk_frequency(freq) != 0) {
-                    return OMV_CSI_ERROR_TIM_INIT_FAILED;
-                }
-                init_ret = ov5640_init(csi);
+        // Find the sensors init function.
+        for (size_t i=0; i<OMV_ARRAY_SIZE(sensor_config_table); i++) {
+            const sensor_config_t *config = &sensor_config_table[i];
+            if (csi->chip_id == config->chip_id) {
+                clk_hz = config->clk_hz;
+                init_fun = config->init_fun;
                 break;
             }
-            #endif // (OMV_OV5640_ENABLE == 1)
-
-            #if (OMV_OV7670_ENABLE == 1)
-            case OV7670_ID:
-                if (omv_csi_set_clk_frequency(OMV_OV7670_CLK_FREQ) != 0) {
-                    return OMV_CSI_ERROR_TIM_INIT_FAILED;
-                }
-                init_ret = ov7670_init(csi);
-                break;
-            #endif // (OMV_OV7670_ENABLE == 1)
-
-            #if (OMV_OV7690_ENABLE == 1)
-            case OV7690_ID:
-                if (omv_csi_set_clk_frequency(OMV_OV7690_CLK_FREQ) != 0) {
-                    return OMV_CSI_ERROR_TIM_INIT_FAILED;
-                }
-                init_ret = ov7690_init(csi);
-                break;
-            #endif // (OMV_OV7690_ENABLE == 1)
-
-            #if (OMV_OV7725_ENABLE == 1)
-            case OV7725_ID:
-                init_ret = ov7725_init(csi);
-                break;
-            #endif // (OMV_OV7725_ENABLE == 1)
-
-            #if (OMV_OV9650_ENABLE == 1)
-            case OV9650_ID:
-                init_ret = ov9650_init(csi);
-                break;
-            #endif // (OMV_OV9650_ENABLE == 1)
-
-            #if (OMV_MT9V0XX_ENABLE == 1)
-            case MT9V0X2_ID_V_1:
-            case MT9V0X2_ID_V_2:
-                // Force old versions to the newest.
-                csi->chip_id = MT9V0X2_ID;
```

Comment:
This is another special case that's missing. It could be moved into mt9v0xx_init() as it's just a chip ID change.

Reply:
I'll update the IDE to handle this.